### DISCUSSION TITLE
Inline Editor

### DISCUSSION THREAD

**Discussion by Zack-Heisnberg** [directus/directus#17937] [Feature Requests]
### Summary

While viewing the table for items  for example this one

![image](https://user-images.githubusercontent.com/69718435/227695829-c0196cba-5f62-4afe-b5a3-0bdac554d0bc.png)

it's would be nice to be able to Enabled and disabled the booleans , also to edit the reserve value directly from the table
without opening the 

### Basic Example

![image](https://user-images.githubusercontent.com/69718435/227696104-079fec07-d721-4b4d-a784-a610e946f9ed.png)

### Motivation

i think it's will bring a huge advantage to directus on general when it's the users need to edit multiple values in a list of items

### Detailed Design

1 - Adding a boolean in the field settings will be called inline_editor
2 - for fields where that boolean is activated the field will use the interface component instead of the display component , ( probably a different compact interfaces for booleans and inputs )
3 - the new display component should listen to changes , once a change is done it's shows a dialog ( are you sure you want t o update the value ? ) 
4 - send the update request and show small notification

### Must Have:
- ( Inputs and Booleans )
- Proper permissions

### Should Have:
- .

### Could Have:
- a Toggle to enable disable in the preset

### Won't Have:
- Images ,  M2A ... etc

### Drawbacks

it's might need lot of changes 

### Alternatives

don't think there is

### Adoption Strategy

the migration will should be simple , just adding  use_interface boolean to fields

**Comment by Zack-Heisnberg** [comment]
implementing a whole layout would be harder i think

https://github.com/directus/directus/discussions/3874

**Comment by Eggwise** [comment]
This is a dealbreaker for me, if i cannot edit inline NoCoDB is a better alternative for the next platform we are building.

Inline editting saves a huge amount of time summing up all the clicks for all the reps that use the tool

**Comment by carwei** [comment]
I like this. But how does it compare with "Spreadsheet Layout" in the Roadmap? That would include inline editing, I assume. But this could be the first iterative step.

**Reply by rijkvanzanten** [reply]
I think the main difference is a presentational style. This "inline editing" would be a way to edit things on the "pretty" displays in a the tabular layout, whereas a spreadsheet is values in a cell? Similar idea, but the UX is a bit different

**Reply by carwei** [reply]
Right. Can an inline editor be implemented in the current version of Directus with a custom _Display_ module? Just want to double-check if this module type's name doesn't represent its full potential 😄

**Reply by rijkvanzanten** [reply]
> Right. Can an inline editor be implemented in the current version of Directus with a custom Display module? Just want to double-check if this module type's name doesn't represent its full potential 😄

Yes! A display is "just" another Vue component, so you can make it do whatever you want 🙂 

As for support in core, the main challenge is figuring out what the "UX Rules" are around editable displays. Right now, we're using displays in all sorts of places (including the header bar title, cards layout labels etc), so the UX might get confusing if some of those are clickable/editable all of a sudden 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by maks2199** [comment]
Btw, example of this feature can be found in Notion: 
![image](https://github.com/directus/directus/assets/55157621/9842ff25-08dc-4ebd-9bb3-8241b0f581ee)
There is also an option to drug and fill items -- very usful in some cases
![image](https://github.com/directus/directus/assets/55157621/72286199-025e-460a-94cf-f64fe0fb381d)

**Comment by garrulinae** [comment]
FWIW, I just started using Directus today and this was the first feature I looked for.

I want an interface like Notion or Clickup, where you can edit the table with dropdowns etc, or open the record in 'full' in a modal popup or it's own page.

It's a different thing to a spreadsheet layout, because many fields would be dropdowns or tags etc, not just text/number editing.

I'm glad this is under review and I'm really keen to see it implemented!

**Comment by phazonoverload** [comment]
We're inviting you to come and have a chat with us about this feature request at our Request Review in our Discord server on March 21. This will help us have some dedicated time to talk about what implementation would have the most sense. Please feel free to join us as https://directus.chat/ (the event is already listed and you can set a reminder)

**Comment by w0kyj** [comment]
Considerations:
- what types of fields should be allowed?
    - text/string/date/integer
    - relational(m2m,o2m,m2o,m2a,...)
- inline edit
    - handle as raw only (not ideal or user friendly)
    - interface support
        - how to handle complex displays in card/calendar/table/...
    - settings and handling for interfaces with inline edit
        - override display?
        - What if you toggle editing on the entire view? 
            - Use displays by default
            - Then turn on "edit" mode in the collection view
            - Then "inline" interfaces will be used instead of displays?
        - Popover vs inline
            - If interfaces have "inline" variants, those could be used inline, falling back to popovers if interfaces don't have an inline variant?
                - like 3 types: (really) inline, popover interface, drawer for complex fields
            - mobile handling required
- other options:
    - implement as a new layout instead of trying to make all existing layouts work

**Comment by SEED-Foundation** [comment]
Suggestion for Enhanced Table Layout Editing

I propose adding inline editing options for fields within the table layout. This feature would allow users to edit fields directly within the table without needing to navigate away from the page.

Implementation Details:

Enable Inline Editing:

Fields can be set as inline-editable through the data model by setting isEditable to true.
This setting should be specifically applicable to fields in the table layout.
Types of Inline Editable Fields:

- Text: Allow direct editing of text strings.
- Numbers (Integer and Float): Support inline editing of numeric fields.
- Many-to-One Relationships: Enable inline editing for fields that establish many-to-one relationships.

Auto-Save Functionality:
Changes made to fields should automatically save and update without the user having to click a save button.
This enhancement would streamline data modifications and improve user experience by simplifying interactions with the table layout.

**Comment by vwasteels** [comment]
Hello, 
is there any update on this feature ? 
thanks
